storage-platform-clients/src/main/java/com/scality/osis/s3/impl/S3Impl.java

```diff
@@ -0,0 +1,72 @@
+package com.scality.osis.s3.impl;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.securitytoken.model.Credentials;
+import com.amazonaws.util.StringUtils;
+import com.scality.osis.s3.S3;
+import okhttp3.HttpUrl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+
+@Component
+public class S3Impl implements S3 {
+
+    private final String s3Endpoint;
+
+    /**
+     * Create a s3 client implementation
+     *
+     * @param s3Endpoint s3 API endpoint
+     */
+    @Autowired
+    public S3Impl(@Value("${osis.scality.s3.endpoint}") String s3Endpoint) {
```

Comment:
Should `osis.scality.s3.endpoint` be part of application.properties?

Reply:
yes, it should be! thanks for pointing it out
the s3c ticket is created https://scality.atlassian.net/browse/S3C-6526